src/test/regress/output/temp_tablespaces.source

```diff
@@ -494,3 +494,302 @@ NOTICE:  ok
+        files_in_temp_files_n boolean
+    ) AS $$
+DECLARE
+    owo int;
+    session_file_ts_name text;
+    temp_table_ts_name text;
+BEGIN
+    CREATE TEMPORARY TABLE tts_baz (a int) DISTRIBUTED BY (a);
+
+    -- See the tablespace where temporary table was created. If there is no
+    -- tablespace for this table, it's in default tablespace.
+    SELECT ts.spcname
+        FROM pg_tablespace ts
+        JOIN pg_class c ON ts.oid = c.reltablespace AND c.relname = 'tts_baz'
+        LIMIT 1
+        INTO temp_table_ts_name;
+
+    DROP TABLE tts_baz;
+
+    -- Find out the tablespace that will be used for files.
+    -- N = gp_session_id % number of tablespaces in GUC.
+    SELECT 'temp_files' || (current_setting('gp_session_id')::int % n_temp_spill_files_tablespaces)
+        INTO session_file_ts_name;
+
+    -- Create the temporary directory.
+    WITH out11 AS (
+        SELECT * FROM (
+            WITH cte(a,b) AS (SELECT a-1, b+1 FROM tts_foo)
+            SELECT * FROM (SELECT * FROM cte UNION ALL SELECT * FROM cte) AS X JOIN tts_bar ON b = c
+            UNION ALL SELECT *, 1, 2 FROM cte
+        ) AS XY
+        JOIN tts_jazz on c = e AND b = f
+    )
+    SELECT count(*) FROM out11 INTO owo;
+
+    -- Start monitoring temporary tablespace usage.
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), 'default', gp_temptablespace_path((SELECT dattablespace FROM pg_database WHERE datname = current_database() LIMIT 1)));
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), 'temp_tables', gp_temptablespace_path((SELECT oid FROM pg_tablespace WHERE spcname = 'temp_tables' LIMIT 1)));
+    PERFORM gp_tablespace_watch_start(gp_execution_dbid(), session_file_ts_name, gp_temptablespace_path((SELECT oid FROM pg_tablespace WHERE spcname = session_file_ts_name LIMIT 1)));
+
+    -- Run the query again to create temporary hashjoin files in the tablespaces.
+    WITH out11 AS (
+        SELECT * FROM (
+            WITH cte(a,b) AS (SELECT a-1, b+1 FROM tts_foo)
+            SELECT * FROM (SELECT * FROM cte UNION ALL SELECT * FROM cte) AS X JOIN tts_bar ON b = c
+            UNION ALL SELECT *, 1, 2 FROM cte
+        ) AS XY
+        JOIN tts_jazz on c = e AND b = f
+    )
+    SELECT count(*) FROM out11 INTO owo;
+
+    PERFORM gp_tablespace_watch_stop();
+
+    -- Check the existence of temporary hashjoin files.
+    RETURN QUERY SELECT
+        coalesce(temp_table_ts_name, 'default'),
+        gp_tablespace_watch_match(gp_execution_dbid(), 'default', '_HashJoin') > 0,
+        gp_tablespace_watch_match(gp_execution_dbid(), 'temp_tables', '_HashJoin') > 0,
+        gp_tablespace_watch_match(gp_execution_dbid(), session_file_ts_name, '_HashJoin') > 0;
+END;
+$$ LANGUAGE plpgsql;
+--
+-- Global GUC value tests:
+-- FIXME: We shouldn't be forced to use \c to update GUCs on segments.
+--
+-- Both files and the table should be in the default tablespace.
+-- start_ignore
+\! gpconfig -r temp_tablespaces
```

Comment:
?

Reply:
These blocks are here to provide a visual clue as to what settings have been changed in the global configuration, similar to tests for SET. I don't think they should be removed.

Reply:
Either put this out of the ignore block, or it's just trash!

Reply:
> Why do you think it's garbage?

Because it is inside the ignore block in the output file.

Reply:
Add some comments before ignore block, like
```sql
reset temp_tablespaces and temp_spill_files_tablespaces in config file
```